### PR TITLE
CDI-596 ProcessAnnotatedType - clarify what happens if both setAnnotatedType() and configurator are used within observer notification

### PR DESCRIPTION
**Pull request by antoinesd**
An IllegalStateException is thrown

### CODE REVIEW

File: api/src/main/java/javax/enterprise/inject/spi/ProcessAnnotatedType.java [hunk 1]
```diff
@@ -24,9 +24,9 @@
  * the declared annotations.
  * </p>
  * <p>
- * Any observer of this event is permitted to wrap and/or replace the {@link javax.enterprise.inject.spi.AnnotatedType}. The
- * container must use the final value of this property, after all observers have been called, to discover the types and read the
- * annotations of the program elements.
+ * Any observer of this event is permitted to wrap and/or replace the {@link javax.enterprise.inject.spi.AnnotatedType} by calling either {@link #setAnnotatedType(AnnotatedType)} or {@link #configureAnnotatedType()}.
+ * If both methods are called in the observer method, an {@link IllegalStateException} is thrown.
```
Comment (mkouba) [inline api/src/main/java/javax/enterprise/inject/spi/ProcessAnnotatedType.java]: If both methods are called within an observer notification...

**Comment by mkouba** [comment]
This should apply to `ProcessObserverMethod`, `ProcessBeanAttributes` and `ProcessInjectionPoint` as well.

**Comment by antoinesd** [comment]
Yes but I wanted to have the correct wording before going for the other 3...
If we are good, I go for it now...

File: api/src/main/java/javax/enterprise/inject/spi/ProcessAnnotatedType.java (hunk 1 above)
Comment (antoinesd) [inline api/src/main/java/javax/enterprise/inject/spi/ProcessAnnotatedType.java]: ok, making the same change in spec wording.

**Comment by antoinesd** [comment]
IMO Should be ready to merge now. WDYT @mkouba?

**Comment by mkouba** [comment]
Yep, looks good.

**Pull request merged**